decam_process.py

```diff
@@ -207,158 +246,307 @@ def doIngest(repo, datafiles):
+    args.extend(calibdatafiles)
     argumentParser = IngestCalibsArgumentParser(name='ingestCalibs')
     config = IngestCalibsConfig()
     config.parse.retarget(ingestCalibs.DecamCalibsParseTask)
-    ingestTask = IngestCalibsTask(config=config, name='ingestCalibs')
+    CalibIngestTask = IngestCalibsTask(config=config, name='ingestCalibs')
     parsedCmd = argumentParser.parse_args(config=config, args=args)
     try:
-        ingestTask.run(parsedCmd)
+        CalibIngestTask.run(parsedCmd)
     except sqlite3.IntegrityError as detail:
-        print('~~~ !!! ~~~')
         print('sqlite3.IntegrityError: ', detail)
         print('(sqlite3 doesn\'t think all the calibration files are unique)')
-        print('If this isn\'t your first time ingesting these calibration')
-        print('  files, move or delete the existing database and try again.')
-        print('If this is your first time ingesting these calibration files,')
-        print('  make sure you only use image or wtmap MasterCals, not both.')
-        print('~~~ !!! ~~~')
+        flatBias_metadata = None
     else:
         print('Success!')
-        print('Calibrations corresponding to {0} are now ingested in {1}'.format(repo, calibrepo))
-    return
+        print('Calibrations corresponding to {0} are now ingested in {1}'.format(repo, calib_repo))
+        flatBias_metadata = CalibIngestTask.getFullMetadata()
+    return flatBias_metadata
 
 
-def doProcessCcd(repo, calibrepo, processedrepo, visit, ccdnum):
+def defectIngest(repo, calib_repo, defectfiles):
     '''
-    Perform ISR with ingested images and calibrations via processCcd
+    Ingest DECam defect images (called by doIngestCalibs)
+
+    Parameters
+    ----------
+    repo: `str`
+        The output repository location on disk where ingested raw images live.
+    calib_repo: `str`
+        The output repository location on disk where ingested calibration images live.
+    defectfiles: `list`
+        A list of the filenames of each defect image file.
+        The first element in this list must be the name of a .tar.gz file
+        which contains all the compressed defect images.
+
+    Returns
+    -------
+    defect_metadata: `PropertySet` or None
+        Metadata from the IngestCalibTask (defects) for use by ap_verify
+
+    BASH EQUIVALENT:
+    $ cd calib_repo
+    $ ingestCalibs.py ../../repo --calib . --mode=skip --calibType defect --validity 999 defectfiles
+    $ cd ..
+
+    This function assumes very particular things about defect ingestion:
+    - They must live in a .tar.gz file in the same location on disk as the other calibs
+    - They will be ingested using ingestCalibs.py run from the calib_repo directory
+    - They will be manually uncompressed and saved in calib_repo/defects/<tarballname>/.
+    - They will be added to the calib registry, but not linked like the flats and biases
+    '''
+    os.chdir(calib_repo)
+    try:
+        os.mkdir('defects')
+    except OSError:
```

Comment:
Never mind, it's always `OSError`. :astonished: 

You'll probably brand this as "not this ticket", but I don't think it's safe to assume a pre-existing directory is the only reason `os.mkdir` raises -- it could also be lack of permissions, problems with the filesystem itself, etc...